examples/chapter-list.lua

```diff
@@ -1,51 +1,117 @@
 --[[
     This script implements an interractive chapter list
+    Usage: add bindings to input.conf
+    -- key script-message-to chapter_list toggle-chapter-browser
 
     This script was written as an example for the mpv-scroll-list api
     https://github.com/CogentRedTester/mpv-scroll-list
 ]]
 
 local mp = require 'mp'
+local opts = require("mp.options")
+
+local o = {
+    -- header of the list
+    -- %cursor% and %total% to be used to display the cursor position and the total number of lists
+    header = "Chapter List [%cursor%/%total%]\\N ------------------------------------",
+    -- wrap the cursor around the top and bottom of the list
+    wrap = true,
+    -- reset cursor navigation when open the list
+    reset_cursor_on_close = true,
+    -- set dynamic keybinds to bind when the list is open
+    key_move_begin = "HOME",
+    key_move_end = "END",
+    key_move_pageup = "PGUP",
+    key_move_pagedown = "PGDWN",
+    key_scroll_down = "DOWN WHEEL_DOWN",
+    key_scroll_up = "UP WHEEL_UP",
+    key_open_chapter = "ENTER MBTN_LEFT",
+    key_close_browser = "ESC MBTN_RIGHT",
+}
+
+opts.read_options(o)
 
 --adding the source directory to the package path and loading the module
 local list = dofile(mp.command_native({"expand-path", "~~/script-modules/scroll-list.lua"}))
 
 --modifying the list settings
-list.header = "Chapter List \\N ------------------------------------"
+local original_open = list.open
+list.header = o.header
+list.wrap = o.wrap
 
---jump to the selected chapter
-local function open_chapter()
-    if list.list[list.selected] then
-        mp.set_property_number('chapter', list.selected - 1)
-    end
+--escape header specifies the format
+--display the cursor position and the total number of lists in the header
+function list:format_header_string(string)
```

Comment:
```suggestion
function list:format_header_string(str)
```
Don't use `string` as a variable, it interferes with the default string library. It's fine here but it introduces confusion.